x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/scheduler/SchedulerEngine.java

```diff
@@ -146,8 +157,13 @@ public int jobCount() {
 
     protected void notifyListeners(String name, long triggeredTime, long scheduledTime) {
         final Event event = new Event(name, triggeredTime, scheduledTime);
-        for (Listener listener : listeners) {
-            listener.triggered(event);
+        for (final Listener listener : listeners) {
+            try {
+                listener.triggered(event);
+            } catch (final Exception e) {
+                // do not allow exceptions to escape this method; we should continue to notify listeners and schedule the next run
+                logger.warn("listener failed while handling triggered event [{}]", name);
```

Comment:
can we log the exception? maybe debug?

Reply:
Good call. I pushed 85283277a14e5721432155db236ad11b63ade548.